src/frequenz/sdk/microgrid/component_graph.py

```diff
@@ -748,6 +773,67 @@ def dfs(
 
         return component
 
+    def find_first_descendant(
+        self,
+        root_category: ComponentCategory = ComponentCategory.GRID,
+        descendant_categories: Iterable[ComponentCategory] | None = None,
```

Comment:
If you really want to keep the default here (I don't think we should), you can also make it more explicit by using a tuple (which is immutable, so it is fine to use as a default argument):

```suggestion
        descendant_categories: Iterable[ComponentCategory] | None = (
            ComponentCategory.METER,
            ComponentCategory.INVERTER,
            ComponentCategory.EV_CHARGER
            )
        )
```

This way it will also show automatically in the generated docs.

Reply:
No, I don't have any strong opinion to keep default values

Reply:
Removed default values